### PR TITLE
LDAP: remove unused methods and DB values

### PR DESCRIPTION
**Pull request by blizzz**
We were still having the `update()` method on LDAP's `User` class as well as related values we stored with the users. In the past we changed updating of user records to happen initially (on first importing), in the background, and on login. The use case to do it inline was not given anymore. `update()` was now only referenced by unit tests. It made use of some other methods and also of the `lastFeatureRequest` marker in the database. This is not needed anymore, and thus we do not need to write it anymore on every background run. The repair steps take care of removing these rows from the database.

### CODE REVIEW

**Review by rullzer — APPROVED** [review]
:rocket:

File: apps/user_ldap/lib/Migration/RmRefreshTime.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+use OCP\Migration\IRepairStep;
+
+/**
+ * Class RmRefreshTime
+ *
+ * this can be removed with Nextcloud 21
+ *
+ * @package OCA\User_LDAP\Migration
+ */
+class RmRefreshTime implements IRepairStep {
```
Comment (juliushaertl) [inline apps/user_ldap/lib/Migration/RmRefreshTime.php:40]:
```suggestion
class RemoveRefreshTime implements IRepairStep {
```

A bit easier to get ;)

**Review by juliusknorr — APPROVED** [review]
Small naming suggestion, otherwise :+1:

File: apps/user_ldap/lib/Migration/RmRefreshTime.php (hunk 1 above)
Comment (blizzz) [reply apps/user_ldap/lib/Migration/RmRefreshTime.php:40]: as you wish

**Review by ChristophWurst — APPROVED** [review]
:+1: :rocket: :broom:

**Pull request merged**